### PR TITLE
fix(inputs.ipset): Parse lines with timeout

### PR DESCRIPTION
**Pull request by powersj**
Parsing raw CLI output, and lines with a timeout shift the values over by two. This checks for timeout, and if it exists adds it as a field as well.

fixes: #14259

### CODE REVIEW

**Comment by powersj** [comment]
> is still potentially quite fragile if things change / other situations arise in the future

I completely agree, which is why I made my comment in the bug about the unfortunate situation of parsing CLI output.

> would probably be okay

Agreed again. I am hesitant to change any logic when the plugin is as old as it is and we have surprisingly not received any issues prior to this but can make the change.

**Comment by attie** [comment]
Looks good to me, thank you!

I hope to test this out tomorrow, and I'll give my thumbs up when confirmed... though I trust your testing.

File: plugins/inputs/ipset/ipset_test.go [hunk 1]
```diff
@@ -74,6 +74,22 @@ func TestIpset(t *testing.T) {
...
+				add counter-test 192.168.1.1 timeout 1792 packets 8 bytes 672
+				create counter-test2 hash:ip family inet hashsize 1024 maxelem 65536 counters
+				add counter-test2 192.168.1.1 packets 18 bytes 673
+				`,
+			tags: []map[string]string{
+				{"set": "counter-test", "rule": "192.168.1.1"},
+				{"set": "counter-test2", "rule": "192.168.1.1"},
+			},
+			fields: [][]map[string]interface{}{
+				{map[string]interface{}{"packets_total": uint64(8), "bytes_total": uint64(672), "timeout": uint64(1792)}},
```
Comment (Hipska) [inline plugins/inputs/ipset/ipset_test.go:89]: Not 1800?
```suggestion
				{map[string]interface{}{"packets_total": uint64(8), "bytes_total": uint64(672), "timeout": uint64(1800)}},
```

Comment (srebhan) [reply plugins/inputs/ipset/ipset_test.go:89]: No. The `create` command defines the **default** value for the set and `add` command can override the value as in this case.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @powersj!

File: plugins/inputs/ipset/ipset_test.go (hunk 1 above)
Comment (Hipska) [reply plugins/inputs/ipset/ipset_test.go:89]: Oh, indeed. Missed that.

**Comment by attie** [comment]
"_tomorrow_" ... works great, many thanks!

**Pull request merged**